### DISCUSSION TITLE
Vertically centered circle in heading

### DISCUSSION THREAD

**Discussion by adrfantini** [typst/typst#4075] [Q&A] [answered]
I'm having issues creating heading with a vertically centered circle in them.
Not sure if it's a bug or intended.

MRE: [link](https://typst.app/project/r7sTBcR9S9IrJHqnzDvsC_)

It seems that:
1. I am forced to use a `box`, otherwise the heading goes to newline. This is somewhat unintuitive
2. even with `align(horizon)`, I still have to manually fiddle with heights to get it right. I would expect that setting `height: 1em` (or something along those lines) would fix it

**Comment by l0wx** [comment, marked as answer]
1. That's because `circle()`  is a block element.
2. Depending on if you want the circle to be in line or to the side:
```typst
= #context box(
    height: measure[.].height,
    align(horizon, circle(height: .3em)),
) #lorem(11)
= #grid(
    align: horizon,
    columns: 3,
    circle(height: .3em),
    [\u{2063} \u{2063}],// Invisible Sep. → Whitespacing trimming mitigation
    lorem(11),
)
```

**Reply by adrfantini** [reply]
Thanks, that works!

It's possibly not that intuitive, to be honest, but it works!

**Reply by l0wx** [reply]
You're welcome! For reasons of clarity and comprehensibility, please mark the answer as accepted if you're satisfied.

**Reply by adrfantini** [reply]
I don't have any button to do so, sorry :(

**Reply by laurmaedje** [reply]
It appears to only work in the Q&A category. I moved the discussion and marked the answer.
